### ISSUE TITLE
Could not load dynamic library 'libcupti.so.11.2'

### ISSUE THREAD

**Issue by yuan0821** [spoonsso/dannce#107]
hi! 
I wonder if anyone meets the error like below. I am not sure if this error affects the result because I get the message that shows the full model has been saved (**Saving full model at end of training**.) 
I think it has not allocated my three A6000 GPU right? thank you!
```
/var/spool/slurmd/job424078/slurm_script: line 13: activate: No such file or directory
2022-05-13 16:56:12.039473: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2022-05-13 16:56:47.960569: I tensorflow/compiler/jit/xla_cpu_device.cc:41] Not creating XLA devices, tf_xla_enable_xla_devices not set
2022-05-13 16:56:47.965506: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcuda.so.1
2022-05-13 16:56:52.460082: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1720] Found device 0 with properties: 
pciBusID: 0000:1b:00.0 name: NVIDIA RTX A6000 computeCapability: 8.6
coreClock: 1.8GHz coreCount: 84 deviceMemorySize: 47.54GiB deviceMemoryBandwidth: 715.34GiB/s
2022-05-13 16:56:52.461279: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudart.so.11.0
2022-05-13 16:56:52.461473: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublas.so.11
2022-05-13 16:56:52.462512: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcublasLt.so.11
2022-05-13 16:56:52.462607: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcufft.so.10
2022-05-13 16:56:52.462673: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcurand.so.10
2022-05-13 16:56:52.465188: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcusolver.so.10'; dlerror: libcusolver.so.10: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /lustre/project/PCI/b152046/final_dannce/lib/python3.7/site-packages/cv2/../../lib64:/opt/share/gsl-1.9/lib:/opt/share/fftw-2.1.5/lib:/usr/local/lib/:
2022-05-13 16:56:52.465309: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcusparse.so.11
2022-05-13 16:56:52.511699: I tensorflow/stream_executor/platform/default/dso_loader.cc:49] Successfully opened dynamic library libcudnn.so.8
2022-05-13 16:56:52.520564: W tensorflow/core/common_runtime/gpu/gpu_device.cc:1757] Cannot dlopen some GPU libraries. Please make sure the missing libraries mentioned above are installed properly if you would like to use GPU. Follow the guide at https://www.tensorflow.org/install/gpu for how to download and setup the required libraries for your platform.
Skipping registering GPU devices...
2022-05-13 16:56:52.529603: I tensorflow/core/platform/cpu_feature_guard.cc:142] This TensorFlow binary is optimized with oneAPI Deep Neural Network Library (oneDNN) to use the following CPU instructions in performance-critical operations:  AVX512F
To enable them in other operations, rebuild TensorFlow with the appropriate compiler flags.
2022-05-13 16:56:52.530515: I tensorflow/compiler/jit/xla_gpu_device.cc:99] Not creating XLA devices, tf_xla_enable_xla_devices not set
2022-05-13 16:56:52.530668: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1261] Device interconnect StreamExecutor with strength 1 edge matrix:
2022-05-13 16:56:52.530727: I tensorflow/core/common_runtime/gpu/gpu_device.cc:1267]      
2022-05-13 16:56:54.432768: I tensorflow/core/profiler/lib/profiler_session.cc:136] Profiler session initializing.
2022-05-13 16:56:54.432919: I tensorflow/core/profiler/lib/profiler_session.cc:155] Profiler session started.
2022-05-13 16:56:54.435204: I tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1365] Profiler found 1 GPUs
2022-05-13 16:56:54.439300: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcupti.so.11.0'; dlerror: libcupti.so.11.0: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /lustre/project/PCI/b152046/final_dannce/lib/python3.7/site-packages/cv2/../../lib64:/opt/share/gsl-1.9/lib:/opt/share/fftw-2.1.5/lib:/usr/local/lib/:
2022-05-13 16:56:54.441853: W tensorflow/stream_executor/platform/default/dso_loader.cc:60] Could not load dynamic library 'libcupti.so'; dlerror: libcupti.so: cannot open shared object file: No such file or directory; LD_LIBRARY_PATH: /lustre/project/PCI/b152046/final_dannce/lib/python3.7/site-packages/cv2/../../lib64:/opt/share/gsl-1.9/lib:/opt/share/fftw-2.1.5/lib:/usr/local/lib/:
2022-05-13 16:56:54.442009: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1415] function cupti_interface_->Subscribe( &subscriber_, (CUpti_CallbackFunc)ApiCallback, this)failed with error CUPTI could not be loaded or symbol could not be found.
2022-05-13 16:56:54.442104: I tensorflow/core/profiler/lib/profiler_session.cc:172] Profiler session tear down.
2022-05-13 16:56:54.442198: E tensorflow/core/profiler/internal/gpu/cupti_tracer.cc:1496] function cupti_interface_->Finalize()failed with error CUPTI could not be loaded or symbol could not be found.
2022-05-13 16:57:39.794409: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
2022-05-13 16:57:39.796346: I tensorflow/core/platform/profile_utils/cpu_utils.cc:112] CPU Frequency: 3200000000 Hz
```

**Comment by yuan0821** [comment]
Here is my dannce environment
```
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main
_openmp_mutex             4.5                       1_gnu
absl-py                   0.15.0                   pypi_0    pypi
astunparse                1.6.3                    pypi_0    pypi
attr                      0.3.1                    pypi_0    pypi
attrs                     21.4.0                   pypi_0    pypi
blas                      1.0                         mkl
brotlipy                  0.7.0           py37h27cfd23_1003
bzip2                     1.0.8                h7b6447c_0
ca-certificates           2022.4.26            h06a4308_0
cachetools                5.0.0                    pypi_0    pypi
certifi                   2021.10.8        py37h06a4308_2
cffi                      1.15.0           py37hd667e15_1
charset-normalizer        2.0.4              pyhd3eb1b0_0
cryptography              37.0.1           py37h9ce1e76_0
cudatoolkit               11.3.1               h2bc3f7f_2
cudnn                     8.2.1                cuda11.3_0
cycler                    0.11.0                   pypi_0    pypi
dannce                    1.2.0                     dev_0    <develop>
dill                      0.3.4                    pypi_0    pypi
ffmpeg                    4.2.2                h20bf706_0
flatbuffers               1.12                     pypi_0    pypi
fonttools                 4.33.3                   pypi_0    pypi
freetype                  2.11.0               h70c0345_0
gast                      0.3.3                    pypi_0    pypi
giflib                    5.2.1                h7b6447c_0
gmp                       6.2.1                h2531618_2
gnutls                    3.6.15               he1e5248_0
google-auth               2.6.6                    pypi_0    pypi
google-auth-oauthlib      0.4.6                    pypi_0    pypi
google-pasta              0.2.0                    pypi_0    pypi
grpcio                    1.32.0                   pypi_0    pypi
h5py                      2.10.0                   pypi_0    pypi
idna                      3.3                pyhd3eb1b0_0
imageio                   2.8.0                    pypi_0    pypi
imageio-ffmpeg            0.4.7                    pypi_0    pypi
importlib-metadata        4.11.3                   pypi_0    pypi
intel-openmp              2021.4.0          h06a4308_3561
jpeg                      9e                   h7f8727e_0
keras-preprocessing       1.1.2                    pypi_0    pypi
kiwisolver                1.4.2                    pypi_0    pypi
lame                      3.100                h7b6447c_0
lcms2                     2.12                 h3be6417_0
ld_impl_linux-64          2.35.1               h7274673_9
libffi                    3.3                  he6710b0_2
libgcc-ng                 9.3.0               h5101ec6_17
libgomp                   9.3.0               h5101ec6_17
libidn2                   2.3.2                h7f8727e_0
libopus                   1.3.1                h7b6447c_0
libpng                    1.6.37               hbc83047_0
libstdcxx-ng              9.3.0               hd4cf53a_17
libtasn1                  4.16.0               h27cfd23_0
libtiff                   4.2.0                h85742a9_0
libunistring              0.9.10               h27cfd23_0
libuv                     1.40.0               h7b6447c_0
libvpx                    1.7.0                h439df22_0
libwebp                   1.2.2                h55f646e_0
libwebp-base              1.2.2                h7f8727e_0
lz4-c                     1.9.3                h295c915_1
markdown                  3.3.7                    pypi_0    pypi
matplotlib                3.5.2                    pypi_0    pypi
mkl                       2021.4.0           h06a4308_640
mkl-service               2.4.0            py37h7f8727e_0
mkl_fft                   1.3.1            py37hd3c417c_0
mkl_random                1.2.2            py37h51133e4_0
multiprocess              0.70.12.2                pypi_0    pypi
ncurses                   6.3                  h7f8727e_2
nettle                    3.7.3                hbbd107a_1
networkx                  2.6.3                    pypi_0    pypi
numpy                     1.19.5                   pypi_0    pypi
oauthlib                  3.2.0                    pypi_0    pypi
opencv-python             4.5.5.64                 pypi_0    pypi
openh264                  2.1.1                h4ff587b_0
openssl                   1.1.1o               h7f8727e_0
opt-einsum                3.3.0                    pypi_0    pypi
packaging                 21.3                     pypi_0    pypi
pillow                    9.0.1            py37h22f2fdc_0
pip                       21.2.2           py37h06a4308_0
protobuf                  3.20.1                   pypi_0    pypi
pyasn1                    0.4.8                    pypi_0    pypi
pyasn1-modules            0.2.8                    pypi_0    pypi
pycparser                 2.21               pyhd3eb1b0_0
pyopenssl                 22.0.0             pyhd3eb1b0_0
pyparsing                 3.0.9                    pypi_0    pypi
pysocks                   1.7.1                    py37_1
python                    3.7.13               h12debd9_0
python-dateutil           2.8.2                    pypi_0    pypi
pytorch                   1.11.0          py3.7_cuda11.3_cudnn8.2.0_0    pytorch
pytorch-mutex             1.0                        cuda    pytorch
pywavelets                1.3.0                    pypi_0    pypi
pyyaml                    6.0                      pypi_0    pypi
readline                  8.1.2                h7f8727e_1
requests                  2.27.1             pyhd3eb1b0_0
requests-oauthlib         1.3.1                    pypi_0    pypi
rsa                       4.8                      pypi_0    pypi
scikit-image              0.19.2                   pypi_0    pypi
scipy                     1.7.3                    pypi_0    pypi
setuptools                62.2.0                   pypi_0    pypi
six                       1.15.0                   pypi_0    pypi
sqlite                    3.38.3               hc218d9a_0
tensorboard               2.9.0                    pypi_0    pypi
tensorboard-data-server   0.6.1                    pypi_0    pypi
tensorboard-plugin-wit    1.8.1                    pypi_0    pypi
tensorflow                2.4.0                    pypi_0    pypi
tensorflow-estimator      2.4.0                    pypi_0    pypi
termcolor                 1.1.0                    pypi_0    pypi
tifffile                  2021.11.2                pypi_0    pypi
tk                        8.6.11               h1ccaba5_1
torchaudio                0.11.0               py37_cu113    pytorch
torchvision               0.12.0               py37_cu113    pytorch
typing-extensions         3.7.4.3                  pypi_0    pypi
urllib3                   1.26.9           py37h06a4308_0
werkzeug                  2.1.2                    pypi_0    pypi
wheel                     0.37.1             pyhd3eb1b0_0
wrapt                     1.12.1                   pypi_0    pypi
x264                      1!157.20191217       h7b6447c_0
xz                        5.2.5                h7f8727e_1
zipp                      3.8.0                    pypi_0    pypi
zlib                      1.2.12               h7f8727e_2
zstd                      1.4.9                haebb681_0

```

**Comment by yuan0821** [comment]
I used the same file as the folder of the demo.

```
io_config not found in io.yaml file, falling back to main config
batch_size not found in io.yaml file, falling back to main config
epochs not found in io.yaml file, falling back to main config
downfac not found in io.yaml file, falling back to main config
lr not found in io.yaml file, falling back to main config
num_validation_per_exp not found in io.yaml file, falling back to main config
max_num_samples not found in io.yaml file, falling back to main config
com_finetune_weights not found in io.yaml file, falling back to main config
crop_height not found in io.yaml file, falling back to main config
crop_width not found in io.yaml file, falling back to main config
com_train_dir set to: ./COM/train_results/
com_predict_dir set to: ./COM/predict_results/
dannce_train_dir set to: ./DANNCE/train_results/AVG/
dannce_predict_dir set to: ./DANNCE/predict_results/
dannce_predict_model set to: ./DANNCE/train_results/AVG/weights.1200-12.77642.hdf5
exp set to: [{'label3d_file': './label3d_demo.mat'}, {'label3d_file': './label3d_demo.mat'}]
io_config set to: io.yaml
batch_size set to: 2
epochs set to: 3
downfac set to: 2
lr set to: 5e-5
num_validation_per_exp set to: 2
max_num_samples set to: 100
com_finetune_weights set to: ./COM/weights/
crop_height set to: [0, 1024]
crop_width set to: [0, 1152]
base_config set to: ../../configs/com_mouse_config.yaml
viddir set to: videos
camnames set to: None
n_channels_out set to: 1
sigma set to: 30
verbose set to: 1
net set to: unet2d_fullbn
gpu_id set to: 0
immode set to: vid
mono set to: False
mirror set to: False
loss set to: mask_nan_keep_loss
num_train_per_exp set to: None
augment_hue set to: False
augment_brightness set to: False
augment_hue_val set to: 0.05
augment_bright_val set to: 0.05
augment_rotation_val set to: 5
data_split_seed set to: None
valid_exp set to: None
dsmode set to: nn
debug set to: False
augment_shift set to: False
augment_zoom set to: False
augment_shear set to: False
augment_rotation set to: False
augment_shear_val set to: 5
augment_zoom_val set to: 0.05
augment_shift_val set to: 0.05
start_batch set to: 0
n_channels_in set to: None
extension set to: None
n_views set to: 6
vid_dir_flag set to: None
chunks set to: None
lockfirst set to: None
load_valid set to: None
drop_landmark set to: None
raw_im_h set to: None
raw_im_w set to: None
n_instances set to: 1
start_sample set to: 0
write_npy set to: None
use_npy set to: False
com_predict_weights set to: None
com_debug set to: None
com_exp set to: None
Using the following *dannce.mat files: ./label3d_dannce.mat
```io_config not found in io.yaml file, falling back to main config
batch_size not found in io.yaml file, falling back to main config
epochs not found in io.yaml file, falling back to main config
downfac not found in io.yaml file, falling back to main config
lr not found in io.yaml file, falling back to main config
num_validation_per_exp not found in io.yaml file, falling back to main config
max_num_samples not found in io.yaml file, falling back to main config
com_finetune_weights not found in io.yaml file, falling back to main config
crop_height not found in io.yaml file, falling back to main config
crop_width not found in io.yaml file, falling back to main config
com_train_dir set to: ./COM/train_results/
com_predict_dir set to: ./COM/predict_results/
dannce_train_dir set to: ./DANNCE/train_results/AVG/
dannce_predict_dir set to: ./DANNCE/predict_results/
dannce_predict_model set to: ./DANNCE/train_results/AVG/weights.1200-12.77642.hdf5
exp set to: [{'label3d_file': './label3d_demo.mat'}, {'label3d_file': './label3d_demo.mat'}]
io_config set to: io.yaml
batch_size set to: 2
epochs set to: 3
downfac set to: 2
lr set to: 5e-5
num_validation_per_exp set to: 2
max_num_samples set to: 100
com_finetune_weights set to: ./COM/weights/
crop_height set to: [0, 1024]
crop_width set to: [0, 1152]
base_config set to: ../../configs/com_mouse_config.yaml
viddir set to: videos
camnames set to: None
n_channels_out set to: 1
sigma set to: 30
verbose set to: 1
net set to: unet2d_fullbn
gpu_id set to: 0
immode set to: vid
mono set to: False
mirror set to: False
loss set to: mask_nan_keep_loss
num_train_per_exp set to: None
augment_hue set to: False
augment_brightness set to: False
augment_hue_val set to: 0.05
augment_bright_val set to: 0.05
augment_rotation_val set to: 5
data_split_seed set to: None
valid_exp set to: None
dsmode set to: nn
debug set to: False
augment_shift set to: False
augment_zoom set to: False
augment_shear set to: False
augment_rotation set to: False
augment_shear_val set to: 5
augment_zoom_val set to: 0.05
augment_shift_val set to: 0.05
start_batch set to: 0
n_channels_in set to: None
extension set to: None
n_views set to: 6
vid_dir_flag set to: None
chunks set to: None
lockfirst set to: None
load_valid set to: None
drop_landmark set to: None
raw_im_h set to: None
raw_im_w set to: None
n_instances set to: 1
start_sample set to: 0
write_npy set to: None
use_npy set to: False
com_predict_weights set to: None
com_debug set to: None
com_exp set to: None
Using the following *dannce.mat files: ./label3d_dannce.mat

**Comment by yuan0821** [comment]
Hi. I try using pip to install TensorFlow; now it works. thanks!
